platform/ui-next/src/components/ThumbnailList/ThumbnailList.tsx

```diff
@@ -11,19 +11,85 @@ const ThumbnailList = ({
   activeDisplaySetInstanceUIDs = [],
   viewPreset,
   ThumbnailMenuItems,
-}: withAppTypes) => {
+}) => {
+  // Filter thumbnails into list items and thumbnail items
+  const listItems = thumbnails.filter(
+    ({ componentType }) => componentType === 'thumbnailNoImage' || viewPreset === 'list'
+  );
+
+  const thumbnailItems = thumbnails.filter(
+    ({ componentType }) => componentType !== 'thumbnailNoImage' && viewPreset === 'thumbnails'
+  );
+
   return (
     <div
       className="min-h-[350px]"
       style={{
         '--radix-accordion-content-height': '350px',
       }}
     >
+      {/* Thumbnail Items */}
+      <div
+        id="ohif-thumbnail-list"
+        className="ohif-scrollbar bg-bkg-low grid grid-cols-[repeat(auto-fit,_minmax(0,135px))] place-items-start gap-[4px] overflow-y-hidden pt-[4px] pr-[2.5px] pl-[5px]"
+      >
+        {thumbnailItems.map(
```

Comment:
Is there a reason not to do:
{thumbnailItems.map(item => (<Thumbnail {...item} ... any extra calculations here like onClick />

Reply:
i like that, let me do it